source/common/stream_info/stream_info_impl.h

```diff
@@ -134,6 +134,8 @@ struct StreamInfoImpl : public StreamInfo {
 
   MonotonicTime startTimeMonotonic() const override { return start_time_monotonic_; }
 
+  SystemTime emitLogTime() const override { return time_source_.systemTime(); }
```

Comment:
Hm. This is a bit surprising. This method simply returns the current time. It doesn't return the time the log was emitted, so much as it returns the current time and it's called by the logger so at *that* time it returns the time the log was emitted. Perhaps we should expose time_source_ via a simple accessor instead? WDYT?

Reply:
I agree, fixed thanks!